### PR TITLE
Update code intel milestones

### PR DESCRIPTION
**Pull request by macraig**
Draft version of the new milestones, lsif clang testing infrastructure TBD

### CODE REVIEW

**Review by efritz — APPROVED** [review]
backend stability section LGTM

File: handbook/engineering/code-intelligence/goals.md [hunk 1]
```diff
@@ -25,46 +27,110 @@ Progress on adoption and usage is tracked in our [Looker dashboard](https://sour
...
+1. ✅ Sourcegraph Cloud has precise code intelligence for C++ repositories that use cmake as a build system. 
     - ✅ Auto-index 5 repositories on Sourcegraph Cloud that use cmake as a build system.
     - ✅ Research: Add inference rules for cmake projects to auto-index on Cloud.
-    - 🔄 Auto-detect C++/cmake repositories and display common configurations and documentation to guide users in their set-up.
-    
+   
      **Outcome:** 5 popular open source cmake repositories have up-to-date precise code intelligence on Sourcegraph Cloud.
 
+1. Build lsif-clang testing infrastructure
+    - TBD roadmap item
```
Comment (scalabilitysolved) [inline handbook/engineering/code-intelligence/goals.md:37]: If these are TBD does it make sense to move to after the Bazel milestone so we can set the steps out?

**Review by scalabilitysolved — APPROVED** [review]
Looks good :+1:

File: handbook/engineering/code-intelligence/goals.md [hunk 2]
```diff
@@ -25,46 +27,110 @@ Progress on adoption and usage is tracked in our [Looker dashboard](https://sour
...
+1. Increase Java precise code intel monthly operations.
+    - Deliver to [*N<sub>4</sub>*][N4].
+    - Deliver to [*N<sub>5</sub>*][N5].
+    - Deliver to [*N<sub>6</sub>*][N6].
+    
+    **Outcome:** 3 enterprise customers with Java precise code intel configured in their repositories.
+    
+1. Lsif-java supports third party dependency navigation.
+
+1. Lsif-java supports Java 11-16.
```
Comment (Strum355) [inline handbook/engineering/code-intelligence/goals.md:104]: Im not sure which should come first here, or if this isnt a strict ordering. I feel supporting 11+ would be a quicker win than third party navigation, which afaik relies on RFC 307 which is a big task :smile: if we get wider JDK support we might be able to get local intel out to a few more customers/repos and then iterate upon that by working on library navigation. Im not too concerned about this but just thought it put my 2c out there @olafurpg @macraig

File: handbook/engineering/code-intelligence/goals.md (hunk 1 above)
Comment (macraig) [reply handbook/engineering/code-intelligence/goals.md:37]: @tjdevries and @gbrik were just syncing on this when I opened the PR. I'll update with the new steps, but we can change the order anyway if we feel that's the best way forward.

File: handbook/engineering/code-intelligence/goals.md [hunk 3]
```diff
@@ -25,46 +27,110 @@ Progress on adoption and usage is tracked in our [Looker dashboard](https://sour
...
     - ✅ Auto-index 5 repositories on Sourcegraph Cloud that use cmake as a build system.
     - ✅ Research: Add inference rules for cmake projects to auto-index on Cloud.
-    - 🔄 Auto-detect C++/cmake repositories and display common configurations and documentation to guide users in their set-up.
-    
+   
      **Outcome:** 5 popular open source cmake repositories have up-to-date precise code intelligence on Sourcegraph Cloud.
 
+1. Build lsif-clang testing infrastructure
+    - TBD roadmap item
+    - TBD roadmap item
```
Comment (macraig) [inline handbook/engineering/code-intelligence/goals.md:38]:
```suggestion
    - Build a tool which can to semantically compare and diff LSIF dumps.
    - Write a test harness for lsif-clang which can perform snapshot tests on small code snippets.
    - Spike: Explore how to perform snapshot tests on large projects for integration testing.

```

Comment (macraig) [inline handbook/engineering/code-intelligence/goals.md]:
```suggestion

```

File: handbook/engineering/code-intelligence/goals.md [hunk 4]
```diff
@@ -25,46 +27,113 @@ Progress on adoption and usage is tracked in our [Looker dashboard](https://sour
...
+1. ✅ Sourcegraph Cloud has precise code intelligence for C++ repositories that use cmake as a build system. 
     - ✅ Auto-index 5 repositories on Sourcegraph Cloud that use cmake as a build system.
     - ✅ Research: Add inference rules for cmake projects to auto-index on Cloud.
-    - 🔄 Auto-detect C++/cmake repositories and display common configurations and documentation to guide users in their set-up.
-    
+   
      **Outcome:** 5 popular open source cmake repositories have up-to-date precise code intelligence on Sourcegraph Cloud.
 
+1. Build lsif-clang testing infrastructure
+    - TBD roadmap item
```
Comment (macraig) [inline handbook/engineering/code-intelligence/goals.md:37]:
```suggestion

```

File: handbook/engineering/code-intelligence/goals.md (hunk 2 above)
Comment (macraig) [reply handbook/engineering/code-intelligence/goals.md:106]: @Strum355 it's a good point, especially considering our conversation about support for 11+ potentially being a quick win. Just to add some context, these are tentatively ordered and we can change them (now or further in the future) if it makes sense. Would you and @olafurpg rather leave it this way or switch it?

**Review by gbrik — APPROVED** [review]
C++ stuff LGTM

File: handbook/engineering/code-intelligence/goals.md [hunk 5]
```diff
@@ -25,46 +27,112 @@ Progress on adoption and usage is tracked in our [Looker dashboard](https://sour
...
+   
      **Outcome:** 5 popular open source cmake repositories have up-to-date precise code intelligence on Sourcegraph Cloud.
 
+1. Build lsif-clang testing infrastructure
+    - Build a tool which can to semantically compare and diff LSIF dumps.
+    - Write a test harness for lsif-clang which can perform snapshot tests on small code snippets.
+    - Spike: Explore how to perform snapshot tests on large projects for integration testing.
+
+    
+    **Outcome:** TBD
```
Comment (macraig) [inline handbook/engineering/code-intelligence/goals.md:42]:
```suggestion
    **Outcome:** We can confidently make code changes to lsif-clang without worrying about obvious bugs.
```

File: handbook/engineering/code-intelligence/goals.md [hunk 6]
```diff
@@ -25,46 +27,112 @@ Progress on adoption and usage is tracked in our [Looker dashboard](https://sour
...
-    - 🔄 Auto-detect C++/cmake repositories and display common configurations and documentation to guide users in their set-up.
-    
+   
      **Outcome:** 5 popular open source cmake repositories have up-to-date precise code intelligence on Sourcegraph Cloud.
 
+1. Build lsif-clang testing infrastructure
+    - Build a tool which can to semantically compare and diff LSIF dumps.
+    - Write a test harness for lsif-clang which can perform snapshot tests on small code snippets.
+    - Spike: Explore how to perform snapshot tests on large projects for integration testing.
+
```
Comment (macraig) [inline handbook/engineering/code-intelligence/goals.md:40]:
```suggestion
```

File: handbook/engineering/code-intelligence/goals.md [hunk 7]
```diff
@@ -25,46 +27,112 @@ Progress on adoption and usage is tracked in our [Looker dashboard](https://sour
...
 
-1. Increase precise code intel monthly operations.
-    - Deliver to [*N<sub>1</sub>*][N1]
-    - Deliver to [*N<sub>2</sub>*][N2]
-    - Deliver to [*N<sub>3</sub>*][N3]
+1. Implement standard C++ code navigation features.
+    - Jump to declaration.
+    - Jump to implementation.
+    - Switch between source and header file.
+
```
Comment (macraig) [inline handbook/engineering/code-intelligence/goals.md:68]:
```suggestion
```

**Comment by tjdevries** [comment]
LGTM @macraig

**Pull request merged**